expressServer/scripts/encryptExistingUsers.js

```diff
@@ -1,38 +1,50 @@
 const mongoose = require('mongoose');
 const dotenv = require('dotenv');
-const app = require('express');
 const crypto = require('crypto');
+const { connectDB, closeDB } = require('../libraries/database.js');
+const { hashPassword } = require('../libraries/hashPassword.js');
 
 dotenv.config();
 
-mongoose.connect(process.env.MONGODB_URI, { useNewUrlParser: true, useUnifiedTopology: true })
-  .catch(err => console.error('Could not connect to MongoDB', err));
-
 const userSchema = new mongoose.Schema({
     username: { type: String, required: true, unique: true },
-    hash: { type: String, required: true },
+    hash: { type: String, required: false },
     salt: { type: String, required: false }
 });
 
 const User = mongoose.model('User', userSchema);
 
-function hashPassword(password, salt) {
-    return crypto.createHash('sha3-512').update(password + salt).digest('hex');
-}
-
 async function encryptAllUserPasswords() {
+    const batchSize = 1000;
+    let totalProcessed = 0;
+
     try {
-        const users = await User.find({});
-        for (let user of users) {
-            const salt = crypto.randomBytes(256).toString('hex');
+        await connectDB();
+
+        while (true) {
+            const users = await User.find({}).limit(batchSize);
+            if (users.length === 0) break
 
-            user.salt = salt;
-            user.hash = hashPassword(user.password, salt);
-            await user.save();
+            for (let user of users) {
+                const salt = crypto.randomBytes(256).toString('hex');
+
+                console.log(user.username);
```

Comment:
This will make your program go slow than if you don't print, instead of printing every user you can print the batch count to know how much items you processed